test/framework/http.go

```diff
@@ -88,13 +103,76 @@ func ExecuteHTTPRequestWithUnmarshalledResponse(namespace, httpMethod, uri, path
 
 // IsHTTPRequestSuccessful makes and checks whether an http request is successful
 func IsHTTPRequestSuccessful(namespace, httpMethod, uri, path, bodyFormat, bodyContent string) (bool, error) {
-	response, err := ExecuteHTTPRequest(namespace, httpMethod, uri, path, bodyFormat, bodyContent)
+	return IsHTTPRequestSuccessfulC(&http.Client{}, namespace, httpMethod, uri, path, bodyFormat, bodyContent)
+}
+
+// IsHTTPRequestSuccessful makes and checks whether an http request is successful using a given client
+func IsHTTPRequestSuccessfulC(client *http.Client, namespace, httpMethod, uri, path, bodyFormat, bodyContent string) (bool, error) {
+	response, err := ExecuteHTTPRequestC(client, namespace, httpMethod, uri, path, bodyFormat, bodyContent)
 	if err != nil {
 		return false, err
 	}
+	io.Copy(ioutil.Discard, response.Body)  // Just read the response to be able to close the connection properly
+	defer response.Body.Close()
 	return checkHTTPResponseSuccessful(namespace, response), nil
 }
 
+// ExecuteHTTPRequestsInThreads executes given number of requests using given number of threads (Go routines).
+// Returns []HTTPRequestResult with the outcome of each thread (HTTPRequestResultSuccess or HTTPRequestResultError).
+// Useful for performance testing.
+func ExecuteHTTPRequestsInThreads(namespace, httpMethod string, requestCount, threadCount int, routeURI, path, bodyFormat, bodyContent string) []HTTPRequestResult {
```

Comment:
other `ExecuteHTTP[...]` are returning an error. it could be the same here instead of panic. and you can deal with it then into the steps

Reply:
No problem, it's not bad to have it unified. I have put the check after the measurement is done to not influence results if it is not needed.